src/lib/compress.c

```diff
@@ -310,3 +310,40 @@ int compress_wait(struct compress *compress, int timeout_ms)
 	return compress->ops->wait(compress->data, timeout_ms);
 }
 
+int compress_get_supported_codecs(unsigned int card, unsigned int device,
+		unsigned int flags, int *codecs, int size)
+{
+	struct compress_ops *ops = &compress_hw_ops;
+	char name[128];
+
+	snprintf(name, sizeof(name), "hw:%u,%u", card, device);
+
+	return ops->get_supported_codecs_by_name(name, flags, codecs, size);
+}
+
+int compress_get_supported_codecs_by_name(const char *name, unsigned int flags, int *codecs, int size)
+{
+	struct compress *compress;
+	int ret = -1;
+
+	compress = calloc(1, sizeof(struct compress));
+	if (!compress)
+		return ret;
+
+	if ((name[0] == 'h') || (name[1] == 'w') || (name[2] == ':')) {
+		compress->ops = &compress_hw_ops;
+	} else {
+		if (populate_compress_plugin_ops(compress, name)) {
+			free(compress);
+			return ret;
+		}
+	}
+
+	ret = compress->ops->get_supported_codecs_by_name(name, flags, codecs, size);
+
+	if (compress->dl_hdl)
+		dlclose(compress->dl_hdl);
```

Comment:
Not sure what is `dlclose()` call here but I notice that for example if we fail earlier this function is not called.

Reply:
That is for when a virtual compress node name is provided? It follows the same pattern in `is_codec_supported_by_name`.

Reply:
I don't get what this does either, and the explanation above does not help.